shared/teams/new-team/index.js

```diff
@@ -11,89 +11,143 @@ import {
   ScrollView,
 } from '../../common-adapters/index'
 import {globalColors, globalMargins, globalStyles, isMobile} from '../../styles'
+import {validTeamnamePart} from '../../constants/teamname'
 
-import type {Props} from './'
+type Props = {
+  baseTeam: string,
+  errorText: string,
+  isSubteam: boolean,
+  joinSubteam: boolean,
+  name: string,
+  onBack: () => void,
+  onJoinSubteamChange: () => void,
+  onNameChange: (n: string) => void,
+  onSetTeamCreationError: (err: string) => void,
+  onSubmit: (fullName: string) => void,
+  pending: boolean,
+}
 
-// This logic is copied from go/protocol/keybase1/extras.go.
+class Contents extends React.Component<Props> {
+  _onSubmit = () => {
+    if (!validTeamnamePart(this.props.name)) {
+      this.props.onSetTeamCreationError(
+        'Teamnames must be between 2 and 16 characters long, can only contain letters and underscores, and cannot begin with an underscore.'
+      )
+      return
+    }
+    this.props.onSubmit(this._fullName())
+  }
 
-const headerText = (baseTeam: string, errorText: string, isSubteam: boolean): string => {
-  if (errorText) {
-    return errorText
+  _headerText = () => {
+    if (this.props.isSubteam) {
+      return `You are creating a subteam of ${this.props.baseTeam}`
+    }
+    return "For security reasons, team names are unique and can't be changed, so choose carefully."
   }
 
-  // TODO: Display an error and disable the submit button if name
-  // isn't a valid teamname.
-  return isSubteam
-    ? `You're creating a subteam of ${baseTeam}.`
-    : "For security reasons, team names are unique and can't be changed, so choose carefully."
-}
+  _fullName = () =>
+    this.props.isSubteam ? this.props.baseTeam.concat(`.${this.props.name}`) : this.props.name
 
-const Contents = ({
-  errorText,
-  baseTeam,
-  isSubteam,
-  joinSubteam,
-  name,
-  onJoinSubteamChange,
-  onNameChange,
-  onSubmit,
-  pending,
-}: Props) => (
-  <ScrollView>
-    <Box style={globalStyles.flexBoxColumn}>
-      <Box style={{...styleContainer, backgroundColor: errorText ? globalColors.red : globalColors.blue}}>
-        <Text
-          style={{margin: globalMargins.tiny, textAlign: 'center', width: '100%'}}
-          type="BodySemibold"
-          backgroundMode={errorText ? 'HighRisk' : 'Announcements'}
-        >
-          {headerText(baseTeam, errorText, isSubteam)}
-        </Text>
-      </Box>
+  render() {
+    const {isSubteam, joinSubteam, name, onJoinSubteamChange, onNameChange, pending} = this.props
+    const errorText = this.props.errorText
+    return (
```

Comment:
Mostly the same, I just added the error text below the header text, the subteam name preview, and tweaked some styles.